runhouse/servers/obj_store.py

```diff
@@ -572,11 +580,40 @@ def delete_for_env_servlet_name(env_servlet_name: str, key: Any):
     def delete_local(self, key: Any):
         self.pop_local(key)
 
+    def delete_env(self, env_name: Any):
+        from runhouse.globals import env_servlets
+
+        # clear keys in the env servlet
+        env_servlet_keys = self.keys_for_env_servlet_name(env_name)
+        self.clear_for_env_servlet_name(env_name)
+
+        # delete the env servlet actor
+        if env_name in env_servlets:
+            actor = env_servlets[env_name]
+            ray.kill(actor)
+
+            del env_servlets[env_name]
+
+        # delete the local key
+        if self.contains_local(env_name):
```

Comment:
I'm confused how this is deleting the env name (it's passing your tests so it seems to be, just confused).

Let's say you're in the server, which has no local storage, which calls `delete_env("env2")`. Then I see that we clear all the keys for `env2`, but wouldn't it not find `env2` in local storage, and therefore `env2` would still  be a key in `base` pointing to `Env(...)`?

Reply:
good point, looks like this branch is basically a no-op that I didn't realize, and the env name is being deleted either way. I'm a little bit confused where exactly that is happening though when `cluster.delete(env_name)` is called -- do you have an idea? bc the env key shouldn't be part of the `keys_for_env_servlet_name` that's cleared.